covid_workplace_model.py

```diff
@@ -0,0 +1,461 @@
+            probability_contacts_sustain : probability that contacts on consecutive days remain
+                                            the same as the previous day. 1 implies that
+                                            all contacts stay the same on consecutive days,
+                                            while 0 implies that previous day's contacts have
+                                            no consequence on next day's contacts.
+            sar_average : average secondary attack rate in range [0, 1]
+            case_rate : daily new cases per population, common ranges are 0.00001 to 0.001
+            num_days : the number of days over which to estimate incidence in the workplace
+            num_sims : number of simulations of stochastic model
+    -----Returns-----
+            An num_sims x num_days array of cumulative incidence for
+                each simulation (along the rows) and each day (along the columns)
+    """
+    logging.debug(
+        f"Estimating cumulative incidence for employees:{employees_at_work}, "
+        f"case_rate:{case_rate}, contacts:{daily_contact_size}, sar:{sar_average}"
+    )
+    betadist_alpha, betadist_beta = get_betadist_parameters(sar_average)
+    daily_contact_size = int(min(daily_contact_size, employees_at_work))
+    rng = np.random.default_rng(42)
+
+    num_infected_gen1 = rng.binomial(
+        n=int(employees_at_work),
+        p=case_rate,
+        size=(num_sims, num_days),
+    )
+    cumulative_incidence = np.zeros((num_sims, num_days), dtype=np.int32)
+    for sim in range(num_sims):
+        # generate new index cases for remaining days of consideration
+        running_sum_infected_gen1 = np.cumsum(num_infected_gen1[sim, :])
+        infected_index_gen1 = rng.choice(
+            int(employees_at_work),
+            size=running_sum_infected_gen1[-1],
+            replace=False,
+            shuffle=False,
+        )
+        sar_gen1 = rng.beta(
+            betadist_alpha, betadist_beta, size=running_sum_infected_gen1[-1]
+        )
+        # estimate number of contacts based on geomteric distribution
+        # for number of failures before first success. Average is daily_contact_size >= 1
+        num_contacts_daily = (
+            rng.geometric(
+                1.0 / (daily_contact_size + 1),
+                size=(running_sum_infected_gen1[-1], num_days),
+            )
+            - 1
+        )
+        previous_contacts = {
+            key: np.empty(0, dtype=np.int32) for key in infected_index_gen1
+        }
+        infected_index_contacts = np.empty(0, dtype=np.int32)
+        for day in range(num_days):
+            if running_sum_infected_gen1[day] > 0:
+                infected_index_gen1_for_day = infected_index_gen1[
+                    : running_sum_infected_gen1[day]
+                ]
+                num_contacts_for_day = num_contacts_daily[
+                    : running_sum_infected_gen1[day], day
+                ]
+
```

Comment:
suggest add a sentence to explain the intent of the following for loop

Reply:
Added.